source/common/http/conn_manager_impl.cc

```diff
@@ -404,6 +415,19 @@ void ConnectionManagerImpl::onIdleTimeout() {
   }
 }
 
+void ConnectionManagerImpl::onConnectionDurationTimeout() {
+  ENVOY_CONN_LOG(debug, "max connection duration reached", read_callbacks_->connection());
+  stats_.named_.downstream_cx_max_duration_reached_.inc();
+  if (!codec_) {
+    // TODO(oleg): figure out if following is true for closing non-idle connections.
```

Comment:
I believe it's fine to do a flush write.  In fact it might be useful to have an onTimeout utility function which can be called from onConectionDurationTimeout and onIdletimeout